src/test/java/org/folio/patronblocks/repository/UserSummaryRepositoryImplTest.java

```diff
@@ -0,0 +1,154 @@
+
+    assertThat(resultUserSummary.getId(), is(userId));
+    assertThat(resultUserSummary.getNumberOfLostItems(), is(2));
+    assertThat(resultUserSummary.getNumberOfOpenFeesFinesForLostItems(), is(1));
+    assertThat(resultUserSummary.getOutstandingFeeFineBalance(), is(BigDecimal.ONE));
+  }
+
+  @Test
+  public void shouldUpdateUserSummary() {
+    String userId = UUID.randomUUID().toString();
+    Future<String> userSumId = userSummaryRepository.saveUserSummary(
+      createUserSummary(userId, new BigDecimal(2), 4, 2));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId.result() != null);
+
+    Future<Boolean> isUpdatedFuture = userSummaryRepository.updateUserSummary(
+      createUserSummary(userId, new BigDecimal(10), 3, 1));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(isUpdatedFuture::result);
+
+    Future<Optional<UserSummary>> userSummaryById =
+      userSummaryRepository.getUserSummaryById(userId);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSummaryById.result().isPresent());
+
+    UserSummary updatedUserSummaryRecord = userSummaryById.result().get();
+    assertThat(updatedUserSummaryRecord.getNumberOfLostItems(), is(3));
+    assertThat(updatedUserSummaryRecord.getNumberOfOpenFeesFinesForLostItems(), is(1));
+    assertThat(updatedUserSummaryRecord.getOutstandingFeeFineBalance(), is(BigDecimal.TEN));
+  }
+
+  @Test
+  public void shouldDeleteUserSummary() {
+    String userId1 = UUID.randomUUID().toString();
+    Future<String> userSumId1 = userSummaryRepository.saveUserSummary(
+      createUserSummary(userId1, new BigDecimal(2), 4, 2));
+    String userId2 = UUID.randomUUID().toString();
+    Future<String> userSumId2 = userSummaryRepository.saveUserSummary(
+      createUserSummary(userId2, new BigDecimal(3), 3, 1));
+    Future<String> userSumId3 = userSummaryRepository.saveUserSummary(
+      createUserSummary(UUID.randomUUID().toString(), new BigDecimal(1), 2, 3));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId1.result() != null);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId2.result() != null);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId3.result() != null);
+
+    Future<List<UserSummary>> userSummaries =
+      userSummaryRepository.getUserSummaries(null, 0, 100);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(userSummaries::result, hasSize(3));
+
+    userSummaryRepository.deleteUserSummary(userId1);
+    userSummaryRepository.deleteUserSummary(userId2);
+
+    Future<List<UserSummary>> userSummaryRemaining =
+      userSummaryRepository.getUserSummaries(null, 0, 100);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(userSummaryRemaining::result, hasSize(1));
```

Comment:
It's better to pass userSummaryRepository.getUserSummaries as a condition to Awaitility.